website/docs/reference/administrative-commands/createIndexes.mdx

```diff
@@ -0,0 +1,134 @@
+
+<!-- Syntax structure of the command -->
+
+<!-- markdownlint-disable MD033 -->
+
+This command accepts a collection name and an array of index specification documents.
+The index specification document specifies the fields to be indexed, the index name, and other options such as uniqueness, partial filter expressions, and expiration times for TTL indexes.
+
+```js
+db.runCommand({
+  createIndexes: '<collection_name>',
+  indexes: [
+    {
+      key: {
+        '<field1>': 1, // ascending
+        '<field2>': -1, // descending
+        '<field3>': 'text' // for full-text search
+        // Add more fields as needed
+      },
+      name: '<index_name>', // string
+      unique: false, // Optional
+      partialFilterExpression: {
+        '<field>': { $gte: '<value>' }
+      }, // Optional
+      sparse: false // Optional
+    },
+    // Text index for full-text search
+    {
+      key: { '<field>': 'text' },
+      name: 'text_index_name'
+    },
+
+    // TTL index for automatic expiry
+    {
+      key: { '<timestamp_field>': 1 },
+      name: '<ttl_index_name>',
+      expireAfterSeconds: 3600 // time to expiry in seconds
+    },
+    // Vector index
+    {
+      key: { '<vector_field>': 'cosmosSearch' },
+      name: '<vector_index_name>',
+      cosmosSearchOptions: {
+        kind: 'vector-hnsw',
+        dimensions: 3,
+        similarity: 'COS'
+      }
+    }
+    // Add more index specifications here
+  ]
+})
+```
+
+<!-- majorly fields such as arguments, modifers, stage, operators, parameters, options supported by the command in a table (if necessary or available). Takes in the following parameters: -->
+
+| Field                     | Type     | Description                                                                                                                                                                                                           |
+| ------------------------- | -------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `createIndexes`           | string   | The name of the collection.                                                                                                                                                                                           |
+| `indexes`                 | array    | An array of index specification documents. Each document defines an index.                                                                                                                                            |
+| `key`                     | document | A single document that contains the field(s) to index and their sort order or type (`1`, `-1`, `"text"`, `cosmosSearch` for vector indexes, etc).                                                                     |
```

Comment:
any special reason for double quoting`"text"` but not `cosmosSearch`? :thinking:

Reply:
@Fashander this is still a pending question :hugs: